File: stanford-algorithms/course-1-divide-and-conquer/week-2/selectionsort/selectionsort.go
package main

import (
	"fmt"
)

func selectionsort(arr []int) []int {
	for i := 0; i < len(arr); i++ {
		min := i
		for j := i; j < len(arr); j++ {
			if arr[j] < arr[min] {
				min = j
			}
		}
		arr[i], arr[min] = arr[min], arr[i]
	}
	return arr
}

func main() {
	arr := []int{34, 2, 123, 1, 3, 4, 6, 8}
	fmt.Printf("Initial:\t\t%v\n", arr)
	selectionsort(arr)
	fmt.Printf("Final:\t\t\t%v", arr)
}

File: stanford-algorithms/course-1-divide-and-conquer/week-2/selectionsort/selectionsort.py
def selectionsort(arr):
    for i in range(len(arr)):
        min = i
        for j in range(i, len(arr)):
            if arr[j] < arr[min]:
                min = j
        arr[min], arr[i] = arr[i], arr[min]
    return arr

def main():
    arr = [134, 6, 24, 1, 3213]
    print(f"Initial:\t\t{arr}")
    selectionsort(arr)
    print(f"Final:\t\t\t{arr}")


if __name__ == "__main__":
    main()